core/src/main/scala/org/apache/spark/executor/CoarseGrainedExecutorBackend.scala

```diff
@@ -213,9 +207,17 @@ private[spark] class CoarseGrainedExecutorBackend(
       logInfo(s"Received tokens of ${tokenBytes.length} bytes")
       SparkHadoopUtil.get.addDelegationTokens(tokenBytes, env.conf)
 
-    case DecommissionSelf =>
-      logInfo("Received decommission self")
+    case DecommissionExecutor =>
       decommissionSelf()
+
+    case ExecutorSigPWRReceived =>
+      decommissionSelf()
+      if (driver.nonEmpty) {
```

Comment:
Doing this after decommissionSelf introduces a race condition, not too important one but given that it wasn't in the original flow I'd suggest swapping the order.

Reply:
What's race condition?

Reply:
So we don’t ask the driver to stop scheduling jobs on us first, and the driver could ask us to run a job while we are part way through decommissioning. This won’t result in a failure because well accept the job but it will slow down the decommissioning. So swap the order of these two.

Reply:
Addressed